README.md

```diff
@@ -1 +1,1205 @@
+[Build]   [main]  ---> Using cache
+[Build]   [main]  ---> 0cdc6d1d7af9
+[Build]   [main] Step 5/9 : COPY . ./
+[Build]   [main]  ---> fec02483c800
+[Build]   [main] Step 6/9 : RUN echo "Rebuild the image"
+[Build]   [main]  ---> Running in 0c1f2cbca19f
+[Build]   [main] Rebuild the image
+[Build]   [main] Removing intermediate container 0c1f2cbca19f
+[Build]   [main]  ---> aec92835242c
+[Build]   [main] Step 7/9 : CMD ["npm", "start"]
+[Build]   [main]  ---> Running in 824e56df2920
+[Build]   [main] Removing intermediate container 824e56df2920
+[Build]   [main]  ---> c49d013039de
+[Build]   [main] Step 8/9 : LABEL io.resin.local.image=1
+[Build]   [main]  ---> Running in e9158ba83571
+[Build]   [main] Removing intermediate container e9158ba83571
+[Build]   [main]  ---> 2317085d9161
+[Build]   [main] Step 9/9 : LABEL io.resin.local.service=main
+[Build]   [main]  ---> Running in ebcaf1a3351d
+[Build]   [main] Removing intermediate container ebcaf1a3351d
+[Build]   [main]  ---> fc5e4459c406
+[Build]   [main] Successfully built fc5e4459c406
+[Build]   [main] Successfully tagged local_image_main:latest
+[Logs]    [9/5/2019, 3:58:56 PM] [main]
+[Logs]    [9/5/2019, 3:58:56 PM] [main] > balena-cli-masterclass@1.0.0 start /usr/src/app
+[Logs]    [9/5/2019, 3:58:56 PM] [main] > node src/helloworld.js; sleep infinity
+[Logs]    [9/5/2019, 3:58:56 PM] [main]
+[Logs]    [9/5/2019, 3:58:56 PM] [main] Hello world!
+```
+Once rebuilt, it will restart the service. Notice that the `Rebuild the image`
+echoed line is now in the build log.
+
+Livepush goes way further than this, however. Only files that affect the
+building of the service force a rebuild. For other files, for example source
+files that run in-service, the Supervisor replaces the files in-situ in the
+relevant container layer.
+To show this, continue to run the `balena push` command and then alter the
+`src/helloworld.js` in a text editor and change:
+```
+console.log('Hello world!');
+```
+to
+```
+console.log('Hello moon!');
+```
+On saving the file, you'll see the following output:
+```
+[Live]    Detected changes for container main, updating...
+[Live]    [main] Restarting service...
+[Logs]    [9/5/2019, 4:02:27 PM] [main]
+[Logs]    [9/5/2019, 4:02:27 PM] [main] > balena-cli-masterclass@1.0.0 start /usr/src/app
+[Logs]    [9/5/2019, 4:02:27 PM] [main] > node src/helloworld.js; sleep infinity
+[Logs]    [9/5/2019, 4:02:27 PM] [main]
+[Logs]    [9/5/2019, 4:02:28 PM] [main] Hello moon!
+```
+Instead of rebuilding the image, which takes time, the file is injected directly
+into the container's filing system and then it is restarted. This happens in a
+few seconds and makes the process of developing much faster and more convenient.
+
+Sometimes an engineer may not want to rebuild code 'on the fly'. For this reason
+`balena push` in Local Mode also has a `--nolive` option which can be passed
+to it. When using this switch, engineers need to re`push` when they want to
+rebuild code.
```

Comment:
Maybe worth talking about the `balena logs` feature for local mode too, for instance
```
balena logs 192.168.1.1 --service main
balena logs 192.168.1.1
balena logs 192.168.1.1 --system --service main
balena logs abdefg.local --service my-service
```

Reply:
Hmm, curious, I thought I had! If not, I definitely meant to! :)

Reply:
Done